easybuild/easyblocks/e/elsi.py

```diff
@@ -0,0 +1,94 @@
+##
+# Copyright 2009-2019 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for ELSI, implemented as an easyblock
+
+@author: Miguel Dias Costa (National University of Singapore)
+"""
+import os
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.modules import get_software_root, get_software_version
+
+
+class EB_ELSI(CMakeMake):
+    """Support for building ELSI."""
+
+    @staticmethod
+    def extra_options():
+        """Define custom easyconfig parameters for ELSI."""
+
+        extra_vars = {
+            'enable_pexsi': [False, "Enable PEXSI solver", CUSTOM],
+        }
+
+        return CMakeMake.extra_options(extra_vars)
+
+    def configure_step(self):
+        """Custom configure procedure for ELSI."""
+
+        self.cfg['separate_build_dir'] = True
+
+        if self.cfg['enable_pexsi']:
+            self.cfg.update('configopts', "-DENABLE_PEXSI=1")
+
+        if self.cfg['runtest']:
+            self.cfg.update('configopts', "-DENABLE_TESTS=1")
+            self.cfg.update('configopts', "-DENABLE_C_TESTS=1")
+            self.cfg['runtest'] = 'test'
+
+        libs = [lib[3:-2] for lib in os.environ['SCALAPACK_STATIC_LIBS'].split(',')]
```

Comment:
What's the `3:-2` about? At least clarify with a comment? Does this work as expected for multiple toolchains?

Reply:
I think the `...STATIC_LIBS...` environment variables are always a comma separated list of static library filenames, this is just a way of converting it to the format expected by ELSI (semi-colon separated and without `lib` and '.a')

I suppose something like `re.findall('lib(.*?)\.a', os.environ['SCALAPACK_STATIC_LIBS']` would be clearer?